saleor/core/jwt.py

```diff
@@ -103,6 +108,49 @@ def get_user_from_payload(payload: Dict[str, Any]) -> Optional[User]:
 
 def get_user_from_access_token(token: str) -> Optional[User]:
     payload = jwt_decode(token)
-    if payload["type"] != JWT_ACCESS_TYPE:
+    if payload["type"] not in [JWT_ACCESS_TYPE, JWT_THIRDPARTY_ACCESS_TYPE]:
         return None
-    return get_user_from_payload(payload)
+    permission_limits = payload.get(PERMISSION_LIMITS_FIELD, None)
+    user = get_user_from_payload(payload)
+    if permission_limits is not None:
+        user.permission_limits = permission_limits  # type: ignore
+    return user
+
+
+def create_access_token_for_app(app: "App", user: "User"):
+    """Create access token for app.
+
+    App can use user jwt token to proceed given operation on the Saleor side.
+    The token which can be used by App has additional field defining the permissions
+    assigned to it. The permissions set is the intersection of user permissions and
+    app permissions.
+    """
+
+    permissions_dict = {
+        enum.codename: enum.name
+        for permission_enum in PERMISSIONS_ENUMS
+        for enum in permission_enum
+    }
```

Comment:
Maby we should create the function for this?
We have similar functions `get_permissions_codename` and `get_permissions_enum_list` in `saleor\core\permissions.py`

Reply:
How would you like to name it?

Reply:
Maby `get_permissions_dict` ?